compiler/src/iree/compiler/Codegen/LLVMCPU/ConvertToLLVM.cpp

```diff
@@ -338,7 +338,7 @@ static InstrumentationEntry acquireInstrumentationEntry(Location loc,
       builder.create<LLVM::GEPOp>(loc, basePtr.getType(), basePtr, offsetIndex,
                                   /*inbounds=*/true);
   Value offsetPtrI64 = builder.create<LLVM::BitcastOp>(
-      loc, LLVM::LLVMPointerType::get(i64Type), offsetPtr);
+      loc, LLVM::LLVMPointerType::get(builder.getContext()), offsetPtr);
```

Comment:
Try dropping the `BitCastOps` and just using the source value of bitcast instead of the result of the bitcast everywhere, i.e. here replace `offsetPtrI64	 with `offsetPtr`.... Though this is the LLVM dialect, so hopefully upstream MLIR LLVM does not have such bit casts....